modules/lvgl/input/lvgl_common_input.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 Fabian Blatz <fabianblatz@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_MODULES_LVGL_LVGL_COMMON_INPUT_H_
+#define ZEPHYR_MODULES_LVGL_LVGL_COMMON_INPUT_H_
+
+#include <lvgl.h>
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct lvgl_common_input_config {
+	const struct device *input_dev;
+	struct k_msgq *event_msgq;
+};
+
+struct lvgl_common_input_data {
+	lv_indev_drv_t indev_drv;
+	lv_indev_t *indev;
+	lv_indev_data_t pending_event;
+};
+
+int register_lvgl_indev_driver(lv_indev_type_t indev_type, const struct device *dev);
+
+#define LVGL_INPUT_EVENT_MSGQ(_inst, _type) event_msgq_##_type##_inst
```

Comment:
I would add some extra `_` between type and inst: `event_msgq_##_type##_##_inst` so it's coherent with other structures), also change the prefix to match the rest here, maybe `lvgl_input_msg...`. Just makes your life easier to find all correlated symbols if you have to.